test/powershell/Language/Operators/SplitOperator.Tests.ps1

```diff
@@ -0,0 +1,236 @@
+Describe "Split Operator" -Tags CI {
+    Context "Binary split operator" {
+        It "Binary split operator can split array of value" {
+            $res = "a b", "c d" -split " "
+            $res.count | Should Be 4
+            $res[0] | Should Be "a"
+            $res[1] | Should Be "b"
+            $res[2] | Should Be "c"
+            $res[3] | Should Be "d"
+        }
+
+        It "Binary split operator can split a string" {
+            $res = "a b c d" -split " "
+            $res.count | Should Be 4
+            $res[0] | Should Be "a"
+            $res[1] | Should Be "b"
+            $res[2] | Should Be "c"
+            $res[3] | Should Be "d"
+        }
+
+        It "Binary split operator can works with max substring limit" {
+            $res = "a b c d" -split " ", 2
+            $res.count | Should Be 2
+            $res[0] | Should Be "a"
+            $res[1] | Should Be "b c d"
+
+            $res = "a b c d" -split " ", 0
+            $res.count | Should Be 4
+            $res[0] | Should Be "a"
+            $res[1] | Should Be "b"
+            $res[2] | Should Be "c"
+            $res[3] | Should Be "d"
+
+            $res = "a b c d" -split " ", -1
```

Comment:
Is this a valid scenario, or should PowerShell report an error when the count is less than 1?

Reply:
I get this from our [documentation](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_split?view=powershell-5.1) - see <Max-substrings>.
So it will be a breaking change if we add a exception here. I don't see the need for that.

@mklement0 Could you please comment too?

Reply:
@iSazonov: Yes, the documentation states "A value of 0 and negative values return all the substrings". 

Supporting `0` definitely makes sense, because in order to be able to specify _options_ you must also specify a `<max-substrings>` value for syntactical reasons, so you need a way to signal the default of "return all".

Negative values serve no real purpose - and could lead people to mistakenly assume that they have special meaning - but given that they're documented and that we could break existing code, I don't think a change is warranted.

Reply:
@mklement0 Thanks for good comment!

Reply:
There is an argument in favor of a breaking change - is a negative count a likely bug? For example, if I compute the count and swapped my operands, this feels like a bug PowerShell should catch.

I do think it's an unlikely breaking change, mostly because the count it most commonly a constant.

Reply:
> if I compute the count and swapped my operands

I don't understand. You mean accidentally swapping the RHS operands?

Regardless, if restricting the "return all" meaning to `0` only is not considered too risky, I welcome it.

It would leave the door open for negative values serving a distinct (yet to be implemented) purpose, such as:

```powershell
> 'a b c d' -split ' ', -1  # only split 1 token _from the end_
a b c
d
```

Reply:
I meant something like:
```powershell
$cnt = $b - $a # Should have been $a - $b
'a b c d' -split ' ', $cnt
```

Reply:
Got it, thanks. Preventing negative values categorically is certainly at odds with what I just proposed, but I'd still consider it an improvement.

Reply:
It is interesting improvement:
```powershell
PS C:\temp> 'a b c d' -split ' ', 2
a
b c d
PS C:\temp> 'a b c d' -split ' ', 1
a b c d
PS C:\temp> 'a b c d' -split ' ', 0
a
b
c
d
PS C:\temp> 'a b c d' -split ' ', -1
a b c d
PS C:\temp> 'a b c d' -split ' ', -2
a b c
d
```

Reply:
@iSazonov: Yes, though your `-2` behavior is what I suggest for `-1` - analogous to index `[-1]` retrieving the last element of a collection.
Not sure if providing a way to not split at all is necessary.

```powershell
> 'a b c d' -split ' ', -1  # last 1
a b c
d

> 'a b c d' -split ' ', -2  # last 2
a b
c
d
```

Also, it's probably worth always returning a first item that represents the remaining, unsplit part - even if that part is empty:

```powershell
> 'a b' -split ' ', -2  # last 2, with nothing remaining
   # empty string  to signal that there's no remaining part
a
b
```

That enables the following idiom, irrespective of whether there's a remaining part or not.

```powershell
> $rest, $tokens = 'a b' -split ' ', -2
```

Reply:
@mklement0  I like the "breaking change". If @lzybkr approve we should open Issue Enhancement.